examples/trl/README.md

```diff
@@ -49,6 +54,51 @@ There are two main steps to the DPO training process:
         --output_dir="dpo" \
         --report_to=none
     ```
+For large model like Llama2-70B, we could use DeepSpeed Zero-3 to enable DPO training in multi-card.
+steps like:
+1. Supervised fine-tuning of the base llama-v2-70b model to create llama-v2-70b-se:
+
+    ```
+    DEEPSPEED_HPU_ZERO3_SYNC_MARK_STEP_REQUIRED=1 python ../gaudi_spawn.py --world_size 8 --use_deepspeed sft.py \
+        --model_name_or_path meta-llama/Llama-2-70b-hf \
+        --deepspeed ../language-modeling/llama2_ds_zero3_config.json \
+        --output_dir="./sft" \
+        --max_steps=500 \
+        --logging_steps=10 \
+        --save_steps=100 \
+        --per_device_train_batch_size=1 \
+        --per_device_eval_batch_size=1 \
+        --gradient_accumulation_steps=2 \
+        --learning_rate=1e-4 \
+        --lr_scheduler_type="cosine" \
+        --warmup_steps=100 \
+        --weight_decay=0.05 \
+        --optim="paged_adamw_32bit" \
+        --lora_target_modules "q_proj" "v_proj" \
+        --bf16 \
+        --remove_unused_columns=False \
+        --run_name="sft_llama2" \
+        --report_to=none \
+        --use_habana \
+        --use_lazy_mode
+    ```
+    To merge the adaptors to get the final sft merged checkpoint, we can use the `merge_peft_adapter.py` helper script that comes with TRL:
+    ```
+    python merge_peft_adapter.py --base_model_name="meta-llama/Llama-2-70b-hf" --adapter_model_name="sft" --output_name="sft/final_merged_checkpoint"
+    ```
```

Comment:
Saw issue as below when running above cmd
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/transformers/generation/configuration_utils.py", line 578, in save_pretrained
    raise ValueError(str([w.message for w in caught_warnings]))
ValueError: [UserWarning('`do_sample` is set to `False`. However, `temperature` is set to `0.9` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `temperature`.'), UserWarning('`do_sample` is set to `False`. However, `top_p` is set to `0.6` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `top_p`.')]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/litang/github/438/optimum-habana/examples/trl/merge_peft_adapter.py", line 48, in <module>
    model.save_pretrained(f"{script_args.output_name}")
  File "/usr/local/lib/python3.10/dist-packages/transformers/modeling_utils.py", line 2326, in save_pretrained
    model_to_save.generation_config.save_pretrained(save_directory)
  File "/usr/local/lib/python3.10/dist-packages/transformers/generation/configuration_utils.py", line 580, in save_pretrained
    raise ValueError(
ValueError: The generation config instance is invalid -- `.validate()` throws warnings and/or exceptions. Fix these issues to save the configuration.


Please check

Reply:
Hi @libinta  We have tested in two fresh envs. it's ok of SFT+merged adapter. From your Traceback, seems do_sample is False. if you check https://huggingface.co/meta-llama/Llama-2-70b-hf/blob/main/generation_config.json. the do_sample is True and SFT finetune will not change the generation config. please check the llama-2-70b model in your cache, see if it's changed.

Reply:
I've not got any issues running the examples. Have you managed to make it work @libinta ?